sdk/core/core-http/src/webResource.ts

```diff
@@ -103,7 +107,8 @@ export class WebResource {
     onUploadProgress?: (progress: TransferProgressEvent) => void,
     onDownloadProgress?: (progress: TransferProgressEvent) => void,
     proxySettings?: ProxySettings,
-    keepAlive?: boolean
+    keepAlive?: boolean,
+    compress?: boolean
```

Comment:
Can we stop adding new constructor params and make an `additionalOptions` typed object bag?

Reply:
Added `AdditionalRequestOptions` type.  At some point in the future we probably want to move all the optional parameters into this type.